File: woodroid/src/main/java/me/gilo/woodroid/data/api/RefundAPI.java
package me.gilo.woodroid.data.api;


import me.gilo.woodroid.models.Refund;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;
import java.util.Map;

public interface RefundAPI {

    @Headers("Content-Type: application/json")
    @POST("orders/{id}/refunds")
    Call<Refund> create(@Path("id") int order_id, @Body Refund body);

    @GET("orders/{id}/refunds/{refund_id}")
    Call<Refund> view(@Path("id") int order_id, @Path("refund_id") int refund_id);

    @GET("orders/{id}/refunds")
    Call<List<Refund>> list(@Path("id") int order_id);

    @DELETE("orders/{id}/refunds/{refund_id}")
    Call<Refund> delete(@Path("id") int order_id, @Path("refund_id") int refund_id);

    @DELETE("orders/{id}/refunds/{refund_id}")
    Call<Refund> delete(@Path("id") int order_id, @Path("refund_id") int refund_id, @Query("force") boolean force);

    @GET("orders/{id}/refunds")
    Call<List<Refund>> filter(@Path("id") int order_id, @QueryMap Map<String, String> filter);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/RefundAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.Refund
import retrofit2.Call
import retrofit2.http.*

interface RefundAPI {

    @Headers("Content-Type: application/json")
    @POST("orders/{id}/refunds")
    fun create(@Path("id") order_id: Int, @Body body: Refund): Call<Refund>

    @GET("orders/{id}/refunds/{refund_id}")
    fun view(@Path("id") order_id: Int, @Path("refund_id") refund_id: Int): Call<Refund>

    @GET("orders/{id}/refunds")
    fun list(@Path("id") order_id: Int): Call<List<Refund>>

    @DELETE("orders/{id}/refunds/{refund_id}")
    fun delete(@Path("id") order_id: Int, @Path("refund_id") refund_id: Int): Call<Refund>

    @DELETE("orders/{id}/refunds/{refund_id}")
    fun delete(@Path("id") order_id: Int, @Path("refund_id") refund_id: Int, @Query("force") force: Boolean): Call<Refund>

    @GET("orders/{id}/refunds")
    fun filter(@Path("id") order_id: Int, @QueryMap filter: Map<String, String>): Call<List<Refund>>

}